### PR TITLE
Register the PropertyInfo extractor from the Symfony Bridge

### PR DESCRIPTION
**Pull request by dunglas**
Allow to extract metadata about properties coming from Doctrine using the new PropertyInfo component of Symfony 2.8.

### CODE REVIEW

File: DependencyInjection/DoctrineExtension.php [hunk 1]
```diff
@@ -396,6 +396,14 @@ protected function ormLoad(array $config, ContainerBuilder $container)
                 $def->addTag('doctrine.event_subscriber');
             }
         }
+
+        if (!class_exists('Symfony\Bridge\Doctrine\PropertyInfo\DoctrineExtractor')) {
+            return;
+        }
+
+        foreach ($config['entity_managers'] as $name => $entityManager) {
+            $this->loadPropertyInfoExtractor($name, $container);
```
Comment (stof) [inline DependencyInjection/DoctrineExtension.php]: don't loop a second time. Do it in ``loadOrmEntityManager``

File: Tests/ContainerTest.php [hunk 2]
```diff
@@ -50,6 +50,8 @@ public function testContainer()
         $this->assertInstanceOf('Symfony\Bridge\Doctrine\CacheWarmer\ProxyCacheWarmer', $container->get('doctrine.orm.proxy_cache_warmer'));
         $this->assertInstanceOf('Doctrine\Common\Persistence\ManagerRegistry', $container->get('doctrine'));
         $this->assertInstanceOf('Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntityValidator', $container->get('doctrine.orm.validator.unique'));
+        $this->assertInstanceOf('Doctrine\Common\Persistence\Mapping\ClassMetadataFactory', $container->get('doctrine.orm.default_entity_manager.metadata_factory'));
```
Comment (stof) [inline Tests/ContainerTest.php]: you need to check for the existence of the extractor

File: DependencyInjection/DoctrineExtension.php [hunk 3]
```diff
@@ -396,6 +396,14 @@ protected function ormLoad(array $config, ContainerBuilder $container)
                 $def->addTag('doctrine.event_subscriber');
             }
         }
+
+        if (!class_exists('Symfony\Bridge\Doctrine\PropertyInfo\DoctrineExtractor')) {
```
Comment (stof) [inline DependencyInjection/DoctrineExtension.php]: you should check first for the PropertyInfo component, as the bridge does not force installing PropertyInfo (but we force installing the bridge)

Comment (dunglas) [inline DependencyInjection/DoctrineExtension.php]: If the component is not installed, those services will be removed as they are private and unused.

File: Tests/ContainerTest.php (hunk 2 above)
Comment (dunglas) [inline Tests/ContainerTest.php]: It's checked the next line.

**Comment by dunglas** [comment]
@stof PR updated

**Comment by dunglas** [comment]
Is it possible to merge this PR?

**Comment by fabpot** [comment]
:+1:

File: DependencyInjection/DoctrineExtension.php [hunk 4]
```diff
@@ -759,6 +766,29 @@ protected function loadOrmCacheDrivers(array $entityManager, ContainerBuilder $c
     }
 
     /**
+     * Loads a property info extractor for each defined entity manager.
+     *
+     * @param string           $entityManagerName
+     * @param ContainerBuilder $container
+     */
+    protected function loadPropertyInfoExtractor($entityManagerName, ContainerBuilder $container)
```
Comment (xabbuh) [inline DependencyInjection/DoctrineExtension.php]: should be `private` imo

File: composer.json [hunk 5]
```diff
@@ -36,6 +36,7 @@
         "doctrine/orm": "~2.3",
         "symfony/yaml": "~2.2|~3.0",
         "symfony/validator": "~2.2|~3.0",
+        "symfony/property-info": "~2.8|~3.0",
```
Comment (xabbuh) [inline composer.json]: What about adding this to the `suggest` section too?

Comment (dunglas) [inline composer.json]: done

**Comment by dunglas** [comment]
@xabbuh's comments fixed.

File: composer.json [hunk 6]
```diff
@@ -36,14 +36,16 @@
...
         "twig/twig": "~1.10",
         "satooshi/php-coveralls": "~0.6.1",
         "phpunit/phpunit": "~4"
     },
     "suggest": {
-        "symfony/web-profiler-bundle": "to use the data collector",
-        "doctrine/orm": "The Doctrine ORM integration is optional in the bundle."
+        "symfony/web-profiler-bundle": "To use the data collector.",
+        "doctrine/orm": "The Doctrine ORM integration is optional in the bundle.",
+        "symfony/property-info": "To use the PropertyInfo integration."
```
Comment (stof) [inline composer.json]: I'm not sure this suggestion makes sense. Saying "install this component to use this component" looks weird. People wanting to use PropertyInfo will install it. And DoctrineBundle does not provide any feature based on the PropertyInfo which would let people wonder whether they want to use PropertyInfo. It only provides the inner parts for PropertyInfo.

**Comment by kimhemsoe** [comment]
Status?

File: composer.json (hunk 5 above)
Comment (dunglas) [inline composer.json]: Finally removed as asked by @stof (I also think it doesn't add any help to suggest this component).

**Comment by dunglas** [comment]
@kimhemsoe finished. Can be merged (according to me).

**Comment by dunglas** [comment]
Sorry to insist, but is it possible to merge this one?

**Comment by mikeSimonson** [comment]
@dunglas Thanks

**Comment by dunglas** [comment]
@mikeSimonson thank you very much for the review and the merge!

**Pull request merged**